test/e2e/cluster_extension_revision_test.go

```diff
@@ -0,0 +1,209 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"slices"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/remotecommand"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ocv1 "github.com/operator-framework/operator-controller/api/v1"
+	"github.com/operator-framework/operator-controller/internal/operator-controller/features"
+	. "github.com/operator-framework/operator-controller/internal/shared/util/testutils"
+	. "github.com/operator-framework/operator-controller/test/helpers"
+)
+
+func TestClusterExtensionRevision(t *testing.T) {
+	SkipIfFeatureGateDisabled(t, string(features.BoxcutterRuntime))
```

Comment:
it would be nice to split this large test into several smaller, focused on particular aspects:

* clusterextension install
* clusterextension update
* workload availability change
* etc.

Also, should we keep already existing test `TestClusterExtensionVersionUpdate` in `test/e2e/single_namespace_support_test.go`?